src/camp/generators/data/agent.pgsql.jinja

```diff
@@ -0,0 +1,273 @@
+{%- endmacro %}
+{%- macro gen_metadata(enum, obj) %}
+{%- set status = 'null' %}
+{%- if obj.status %}
+{%- set status = "'"~obj.status~"'" %}
+{%- endif %}
+{%- set reference = 'null' %}
+{%- if obj.reference %}
+{%- set reference = "'"~obj.reference~"'" %}
+{%- endif %}
+{%- set description = 'null' %}
+{%- if obj.description | replace('\n', '') %}
+{%- set description = "'"~obj.description | replace('\n', '')~"'" %}
+{%- endif %}
+CALL SP__insert_obj_metadata({{enum}}, '{{obj.name}}', r_data_model_id, {{obj.enum}}, {{status}}, {{reference}}, {{description}}, {{obj.name}}_oid);
+{%- endmacro%}
+-- -------------------------------------------------------------------
+--
+-- File Name: {{adm.name}}.sql
+--
+-- Description: auto generated postgressql for {{adm.name}} 
+--
+-- Notes: TODO
+--
+-- Assumptions: TODO
+--
+-- Modification History:
+-- YYYY-MM-DD    AUTHOR          DESCRIPTION
+-- ----------    --------------  ------------------------------------
+--     AUTO            Auto-generated SQL file
+--
+-- ------------------------------------------------------------------
+
+DO
+$do$
+DECLARE
+ap_spec_id INTEGER;
+fp_spec_id INTEGER;
+r_data_model_id INTEGER;
+{%- if adm.ident %}
+{%- for obj in adm.ident %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.const %}
+{%- for obj in adm.const %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.ctrl %}
+{%- for obj in adm.ctrl %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.edd %}
+{%- for obj in adm.edd %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.oper %}
+{%- for obj in adm.oper %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.sbr %}
+{%- for obj in adm.sbr %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.tbr %}
+{%- for obj in adm.tbr %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.var %}
+{%- for obj in adm.var %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+{%- if adm.typedef %}
+{%- for obj in adm.typedef %}
+{{obj.name}}_oid INTEGER;
+{{obj.name}}_fid INTEGER;
+{{obj.name}}_aid INTEGER;
+{%- endfor %}
+{%- endif %}
+BEGIN
+
+{% set null_string = 'null' %}
+--namespace 
+CALL SP__insert_data_model('ADM', '{{adm.name}}', {{adm.enum}}, {{"'"~adm.revision~"'" if adm.revision else null_string}} , {{"'"~adm.description~"'" if adm.description else null_string}}, r_data_model_id);
+
+{%- if adm.ident %}
+{%- for obj in adm.ident %}
+{{gen_metadata(-1, obj)}}
+{%- if obj.parameters.items|length > 0 %}
```

Comment:
The condition can just be simplified to "if container" which will be falsy if empty.